packages/eslint-plugin-orbit-internal/src/rules/unique-id.ts

```diff
@@ -0,0 +1,42 @@
+import { Rule } from "eslint";
+import { types as t } from "@babel/core";
+
+export default {
+  meta: {
+    type: "problem",
+    docs: {
+      description: "Prevents id namespace collisions",
+      category: "Possible Errors",
+      recommended: true,
+    },
+  },
+
+  create: (context: Rule.RuleContext) => {
+    const JSXAttributes: t.JSXAttribute[] = [];
+
+    return {
+      JSXAttribute(node: t.JSXAttribute) {
+        if (node.value) {
+          if (node.name.name === "id") {
+            JSXAttributes.push(node);
+          }
+        }
+      },
+
+      "Program:exit": () => {
+        JSXAttributes.forEach(n => {
```

Comment:
Just curious, would it be different if you put the logic that's inside this loop in the place of the `JSXAttributes.push(node)` above?